### PR TITLE
Threadpool loops

### PR DESCRIPTION
**Pull request by t2b3**

### CODE REVIEW

**Comment by aardappel** [comment]
this PR is based on the other one. Either make them independent PRs if you want me to review, or wait till the other one is submitted and rebase before submitting

**Comment by t2b3** [comment]
> this PR is based on the other one. Either make them independent PRs if you want me to review, or wait till the other one is submitted and rebase before submitting

Ok I rebased

File: src/document.h [hunk 1]
```diff
@@ -1737,28 +1737,32 @@ struct Document {
...
+                ThreadPool pool(std::thread::hardware_concurrency());   
                 loopallcellssel(c, true) {
-                    if(c->text.image) {
-                        switch(k) {
-                            case A_SAVE_AS_JPEG: {
-                                if(c->text.image->image_type == 'I') {
-                                    wxImage im = ConvertBufferToWxImage(c->text.image->image_data, wxBITMAP_TYPE_PNG);
-                                    c->text.image->image_data = ConvertWxImageToBuffer(im, wxBITMAP_TYPE_JPEG);
-                                    c->text.image->image_type = 'J';
+                    pool.enqueue([k](Cell *c) {
```
Comment (aardappel) [inline src/document.h:1743]: You're starting a thread pool item for every cell, even cells that have images in them.

But regardless, I think this should not be merged at all.
Note that threading in C++ is extremely treacherous, and can lead to subtle bugs. Any code that is called from threads may lead to future bugs. As such, it should only be used where there's great benefit, such as loading time increases. People converting the format between JPEG and PNG is a highly niche operation that occurs infrequently, and is done only once, and is ok to take a little longer.

Comment (t2b3) [reply src/document.h:1743]: Thanks for your feedback!

**Pull request closed without merging**